File: beginner/__1000__/1040/1040.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main(){

    float n1, n2, n3, n4;
    cin >> n1 >> n2 >> n3 >> n4;

    float media = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / 10;

    cout << "Media: " << fixed << setprecision(1) << media << endl;

    if (media >= 7.0){

        cout << "Aluno aprovado." << endl;
    }
    else if (media < 5.0){

        cout << "Aluno reprovado." << endl;
    }
    else{
        float exame, mediaComExame;
        cin >> exame;

        cout << "Aluno em exame." << endl;
        cout << "Nota do exame: " << exame << endl;

        mediaComExame = (media + exame) / 2;

        if (mediaComExame >= 5.0){
            cout << "Aluno aprovado." << endl;
        }
        else{
            cout << "Aluno reprovado." << endl;
        }

        cout << "Media final: "  << fixed << setprecision(1) << mediaComExame << endl;
    }

    return 0;
}

File: beginner/__1000__/1040/1040.rs
use std::io;

fn main(){

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    
    let notas: Vec<f32> = input.trim().split_whitespace().map(|x| x.parse().unwrap()).collect();
    let media = ((notas[0] * 2.0) + (notas[1] * 3.0) + (notas[2] * 4.0) + (notas[3] * 1.0)) / 10.0;
    
    println!("Media: {:.1}", media);
    
    if media >= 7.0 { println!("Aluno aprovado."); } else if media < 5.0 { println!("Aluno reprovado."); }
    else {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let exame: f32 = input.trim().parse().unwrap();

        println!("Aluno em exame.");
        println!("Nota do exame: {:.1}", exame);

        let media_com_exame = (media + exame) / 2.0;

        if media_com_exame >= 5.0 { println!("Aluno aprovado."); } 
        else if media < 5.0 { println!("Aluno reprovado."); }

        println!("Media final: {:.1}", media_com_exame);
    }
}